### PR TITLE
Clone GitHub project

### PR DESCRIPTION
**Pull request by cfierro94**

### CODE REVIEW

**Comment by xZevalx** [comment]
Some small fixes:
- Change class name to something like "GitSettings", since this class has nothing to do with GitHub but with Git.
- Change setDir, handleOk method to private.

File: jswingripples/src/main/java/org/incha/ui/GitSettings.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    private void setDir(File f){
+        this.path = f;
+    }
+    void handleOk() throws IOException {
+        remoteUrl = url.getText();
+        File localPath = null;
+        try {
+            localPath = File.createTempFile("GiHubProject", "", path);
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (sgioia9) [inline jswingripples/src/main/java/org/incha/ui/GitSettings.java]: if `File.createTempFile("GiHubProject", "", path);` throws IOException, then the exception will be caught
and `localPath` will remain null.
By the way, I think it's confusing that `xxxPath` is a File, and not a String.

File: jswingripples/src/main/java/org/incha/ui/GitSettings.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        }
+        if(!localPath.delete()) {
```
Comment (sgioia9) [inline jswingripples/src/main/java/org/incha/ui/GitSettings.java]: Risk of NPE here.

File: jswingripples/src/main/java/org/incha/ui/GitSettings.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+import org.eclipse.jgit.api.Git;
+
+/**
+ * Created by constanzafierro on 22-11-16.
+ */
+public class GitSettings extends JPanel {
+    private JTextField url = new JTextField(20);
+    private JButton select = new JButton("Select");
+    private SourcesEditor sourcesEditor;
+    private File path;
```
Comment (sgioia9) [inline jswingripples/src/main/java/org/incha/ui/GitSettings.java]: I think path is a confusing name for a `File`

File: jswingripples/src/main/java/org/incha/ui/GitSettings.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        final JLabel selected = new JLabel("", JLabel.LEFT);
+        center.add(dirSelect,BorderLayout.WEST);
+        center.add(selected);
+        select.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                File f = sourcesEditor.selectFile();
+                if(f!=null) {
+                    selected.setText(f.toString());
+                    setDir(f);
```
Comment (sgioia9) [inline jswingripples/src/main/java/org/incha/ui/GitSettings.java]: IMO, `setDir(f)` is obscuring the code.
You could just write:
`this.path = f`
especially considering `path` isn't really a member of the GitSettings class, but a somewhat convenient global variable, it seems awkward to create a private setter for it.

File: jswingripples/src/main/java/org/incha/ui/buildpath/AbstractBuildPathEditor.java [hunk 5]
```diff
@@ -159,6 +159,21 @@ protected void addFile() {
...
+    /**
+     * Use the file chooser to select a file
+     * @return the file selected
+     */
+    public File selectFile(){
+        //create file chooser
+        final JFileChooser chooser = createFileChooser();
+
+        if (chooser.showOpenDialog(this) == JFileChooser.APPROVE_OPTION) {
+            final File f = chooser.getSelectedFile();
```
Comment (sgioia9) [inline jswingripples/src/main/java/org/incha/ui/buildpath/AbstractBuildPathEditor.java]: If the dialog is canceled, will f be null? 
If so, you can just return f and obtain the same behavior.

Comment (cfierro94) [inline jswingripples/src/main/java/org/incha/ui/buildpath/AbstractBuildPathEditor.java]: yes it'll be null, and I just return f

File: jswingripples/src/main/java/org/incha/ui/JSwingRipplesApplication.java [hunk 6]
```diff
@@ -153,6 +153,56 @@ protected void importProject(){
     }
 
     /**
+     * Import project from a url in github
+     */
+    protected void importProjectGithub(){
```
Comment (sgioia9) [inline jswingripples/src/main/java/org/incha/ui/JSwingRipplesApplication.java]: I don't think the methods
`importProjectGithub()`
and
`showSettingsGitHub(JavaProject project)`
should be in JSwingRipplesApplication.
Can we move them somewhere else?

File: jswingripples/src/main/java/org/incha/ui/GitSettings.java (hunk 4 above)
Comment (cfierro94) [inline jswingripples/src/main/java/org/incha/ui/GitSettings.java]: I can't because is inside the actionlistener

Comment (sgioia9) [inline jswingripples/src/main/java/org/incha/ui/GitSettings.java]: What about
`path = f`

Comment (cfierro94) [inline jswingripples/src/main/java/org/incha/ui/GitSettings.java]: is not possible (cannot resolve symbol), is a different class

Comment (sgioia9) [inline jswingripples/src/main/java/org/incha/ui/GitSettings.java]: asd
Doesn't that work?

File: jswingripples/src/main/java/org/incha/ui/JSwingRipplesApplication.java (hunk 6 above)
Comment (sgioia9) [inline jswingripples/src/main/java/org/incha/ui/JSwingRipplesApplication.java]: I'd move them to GitSettings, passing the dependencies around.

File: jswingripples/src/main/java/org/incha/ui/GitSettings.java (hunk 4 above)
Comment (cfierro94) [inline jswingripples/src/main/java/org/incha/ui/GitSettings.java]: yeah, you're right, I've changed the name that's why I had the cannot resolve symbol jajja 😅

File: jswingripples/src/main/java/org/incha/ui/GitSettings.java [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+import org.eclipse.jgit.api.Git;
+
+
+public class GitSettings extends JPanel {
+    private JTextField url = new JTextField(20);
+    private JButton select = new JButton("Select");
+    private SourcesEditor sourcesEditor;
+    private File selectedFile;
+    private String remoteUrl;
```
Comment (sgioia9) [inline jswingripples/src/main/java/org/incha/ui/GitSettings.java]: No reason for `remoteUrl` to be global

File: jswingripples/src/main/java/org/incha/ui/GitSettings.java [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+                f.dispose();
+            }
+        });
+        okPanel.add(ok);
+        panel.add(okPanel, BorderLayout.SOUTH);
+        return panel;
+    }
+
+    void handleOk() throws IOException {
+        remoteUrl = url.getText();
```
Comment (sgioia9) [inline jswingripples/src/main/java/org/incha/ui/GitSettings.java]: `String remoteUrl = url.getText();`

File: jswingripples/src/main/java/org/incha/ui/GitSettings.java (hunk 7 above)
Comment (sgioia9) [inline jswingripples/src/main/java/org/incha/ui/GitSettings.java]: No reason for remoteUrl to be global

File: jswingripples/src/main/java/org/incha/ui/GitSettings.java [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+                }
+                f.dispose();
+            }
+        });
+        okPanel.add(ok);
+        panel.add(okPanel, BorderLayout.SOUTH);
+        return panel;
+    }
+
+    void handleOk() throws IOException {
```
Comment (sgioia9) [inline jswingripples/src/main/java/org/incha/ui/GitSettings.java]: You can make this method private now.

**Pull request merged**